### DISCUSSION TITLE
when resizing the terminal visidaa is in,  the sheet not take up all of the available space?

### DISCUSSION THREAD

**Discussion by emilBeBri** [saulpw/visidata#1635] [Q&A]
Hi!

so when I open visidata, the screen will take up the area available to it. However, if I then go to an empty workspace, so visidata can take up the whole area, it does not resize, but keeps the same size, so that the rest of the terminal it's in is just blanck. 

Am I missing something? It makes no sense! I tried ctrl+L to clear screen, does not help.

I'm on i3 manjaro linux, fully updated.

**Comment by saulpw** [comment]
Does `g_` resize all columns to their full width in this case?

**Comment by emilBeBri** [comment]
Nope - it does not seem like columns are compressed in any way, and i got like 20 cols. Rather, visidata just does not seem to use all the space available to it. When resizing max, as you suggest, it uses the same 1/3 of the actual available screen as when not (which means there are fewer columns to see because the ones that are there takes more space )

Here is a link to a screenshot:

https://www.dropbox.com/s/vicu072pr0oowe6/visidata-trouble-scrot_2022-12-26_10-24-27.png?dl=0

**I think I understand when it's happening. I'm starting visidata from within a function in R , in order to manipulate a data.frame, and then when I close it, it gets imported back into the R environment (using temporary csv files). Then this behaviour arises.**

This is the call that get's sent to the system:


```
alacritty  -e \"visidata\" /tmp/RtmpxDoAHS/file37be49ed6aa9.csv

```  
I also tried using kitty  In kitty it was the same as alacritty.  However then using kitty with this arg:

```
 --start-as "maximized"
```


I got this message: 
```

[360 14:27:01.400223] Ignoring resize request for tiny size: 0x0
```


I also tried  xterm.
In xterm it used more space horizontally, but LESS space vertically. It also displayed this  message in the R terminal (e.g. the terminal that spawns the terminal with visidata in it)

```
xterm: invalid window-id 1370
Xlib: sequence lost (0x10000 > 0x18b) in reply type 0x16!

```

So it seems be something about how the terminal that is spawned perceives the amount of pixels available, something like that (I don't have the language to describe this, I"m hoping you do )

It's an important part of my workflow (before i used libreoffice for this but I'd much rather use visidata), so I would be interested in solving this, let me know how I can assist further in nailing this!


  I tried making a window rule in i3 so that visidata would be immediately moved to a separate workspace, but that did not help.   


version I have (newest on AUR): 
saul.pw/VisiData v2.10.2
R 4.2.2

thank you for helping me! :)

**Reply by saulpw** [reply]
Hm, I'm afraid I don't know!  I use i3 myself with kitty and urxvt and it's always worked fine to maximize/resize.  One thing you could try as a hack, is to set `COLUMNS=200` or whatever the width of your fullscreen is, which forcibly overrides the screen width (and `LINES` for height).

**Comment by emilBeBri** [comment]
Ok, maybe someone else using R will find this at some point and help me figure it out, this is above my skill level.  As I wrote, it works fine when calling visidata from a terminal, but when wrapping it in an R function, it happens. This is the key to figurering it out, I think (writing it again to make sure anybody reading this will get this important part )

How to I use that command? Pressing space and writting it gives the message 'no command COLUMNS=200'

**Reply by saulpw** [reply]
It's an environment variable in your shell.  So like `COLUMNS=200 vd foo.csv`.

**Reply by emilBeBri** [reply]
Ah ok. Well, I'm spawning the terminal from within another terminal, , and then you can't  set environmental variables, as far as i can tell. 

so this works:

```
kitty  -e visidata /path/to/file.csv
```
but this doesn't :: 

```
kitty  -e COLUMNS=200 visidata /path/to/file.csv
```

setting it in the .bashrc file does not work either, because it is dynamically updated or something. It's the same for both kitty, xterm, urxvt and alacritty. 
 
Do you know of a way to do that? setting an environmental variable with the -e flag / --comand flag?

**Reply by saulpw** [reply]
Put `COLUMNS=200` first:

```COLUMNS=200 kitty  -e visidata /path/to/file.csv```

**Reply by emilBeBri** [reply]
> Put `COLUMNS=200` first:
> 
> `COLUMNS=200 kitty -e visidata /path/to/file.csv`

This solved it.  thx
